### PR TITLE
Add payment_reference filter to the invoices.list and to the response

### PR DESCRIPTION
**Pull request by dcsg**
Add the possibility to filter by `payment_reference`.
Return the `payment_reference` in the response.

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -2062,6 +2062,7 @@ Get a list of invoices.
                     + outstanding
                     + matched
             + updated_since: `2016-02-04T16:44:33+00:00` (string, optional)
+            + payment_reference: `2019 / 1` (string, optional)
```
Comment (jenssegers) [inline apiary.apib]: PP: Use the same payment reference as in the list example

Comment (jenssegers) [reply apiary.apib]:
```suggestion
            + payment_reference: `+++084/2613/66074+++` (string, optional)
```

File: src/05-invoicing/invoices.apib [hunk 2]
```diff
@@ -19,6 +19,7 @@ Get a list of invoices.
                     + outstanding
                     + matched
             + updated_since: `2016-02-04T16:44:33+00:00` (string, optional)
+            + payment_reference: `2019 / 1` (string, optional)
```
Comment (dcsg) [inline src/05-invoicing/invoices.apib]:
```suggestion
            + payment_reference: `+++084/2613/66074+++` (string, optional)
```

**Review by Yarn-e — DISMISSED** [review]

**Review by jenssegers — DISMISSED** [review]

**Review by luisaterrabrito — DISMISSED** [review]

**Review by steftrenson — DISMISSED** [review]

**Review by EOSullivanBerlin — DISMISSED** [review]

**Review by jdrieghe — DISMISSED** [review]

**Comment by jdrieghe** [comment]
@dcsg can you resolve the conflict here?

**Review by jdrieghe — DISMISSED** [review]

File: src/05-invoicing/invoices.apib [hunk 3]
```diff
@@ -54,6 +55,7 @@ Get a list of invoices.
                 + paid: true (boolean)
                 + paid_at: `2016-02-04T16:44:33+00:00` (string, nullable)
                 + purchase_order_number: `000023` (string, nullable)
+                + payment_reference: `+++084/2613/66074+++` (string)
```
Comment (lowiebenoot) [inline src/05-invoicing/invoices.apib]: Isn't it nullable as well?

Comment (jdrieghe) [reply src/05-invoicing/invoices.apib]: In the end, it shouldn't be anymore. We're filling them up and making sure they're always filled in. The `nullable` can be removed in a migration later.

Comment (jdrieghe) [reply src/05-invoicing/invoices.apib]: Oh, it has to stay `null` for draft invoices maybe?

Comment (dcsg) [reply src/05-invoicing/invoices.apib]: Indeed, forgot that. Will change it 👍

**Review by jdrieghe — APPROVED** [review]

**Review by lowiebenoot — APPROVED** [review]

**Review by EOSullivanBerlin — APPROVED** [review]

**Review by luisaterrabrito — APPROVED** [review]

File: apiary.apib [hunk 4]
```diff
@@ -2063,6 +2063,7 @@ Get a list of invoices.
                     + matched
             + updated_since: `2016-02-04T16:44:33+00:00` (string, optional)
             + purchase_order_number: `000023` (string, optional)
+            + payment_reference: `+++084/2613/66074+++` (string, optional)
```
Comment (stivni) [inline apiary.apib:2066]: Q : why is this optional?

Comment (stivni) [reply apiary.apib:2066]: nvm, I didn't see it was a filter 🙈

**Pull request merged**